volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,447 @@
+            knode_class_off = vmlinux.get_type("device_private").relative_child_offset(
+                "knode_class"
+            )
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device_private = vmlinux.object(
+                    object_type="device_private",
+                    offset=klist_node.vol.offset - knode_class_off,
+                    absolute=True,
+                )
+                yield device_private.device.dereference()
+        else:
+            knode_class_off = vmlinux.get_type("device").relative_child_offset(
+                "knode_class"
+            )
+            for klist_node in klist_devices.k_list.to_list(
+                vmlinux.symbol_table_name + constants.BANG + "klist_node", "n_node"
+            ):
+                device = vmlinux.object(
+                    object_type="device",
+                    offset=klist_node.vol.offset - knode_class_off,
+                    absolute=True,
+                )
+                yield device
+
+    def _get_name(self, vmlinux, block_device, gendisk):
+        # block_device does not have bd_partno before 5.11
+        if vmlinux.get_type("block_device").has_member("bd_partno"):
+            if block_device.bd_partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    block_device.bd_partno
+                )
+        else:
+            if gendisk.part0.partno:
+                return utility.array_to_string(gendisk.disk_name) + str(
+                    gendisk.part0.partno
+                )
+
+        if utility.array_to_string(gendisk.disk_name).startswith("dm-"):
+            if vmlinux.has_type("mapped_device"):
+                mapped_device = vmlinux.object(
+                    object_type="mapped_device",
+                    offset=gendisk.private_data,
+                    absolute=True,
+                )
+
+                hash_cell = vmlinux.object(
+                    object_type="hash_cell",
+                    offset=mapped_device.interface_ptr,
+                    absolute=True,
+                )
+
+                if hash_cell.name:
+                    return utility.pointer_to_string(hash_cell.name, 32)
+
+        return utility.array_to_string(gendisk.disk_name)
+
+    def _format_bytes(self, size):
+        """
+        Convert a byte value into a human-readable format.
+
+        :param size: Size in bytes
+        :return: Human-readable string
+        """
+
+        if size < 1024:
+            return f"{size}B"
+        elif size < 1024**2:
+            return f"{size / 1024:.1f}K"
+        elif size < 1024**3:
+            return f"{size / 1024 ** 2:.1f}M"
+        elif size < 1024**4:
+            return f"{size / 1024 ** 3:.1f}G"
+        return f"{size / 1024 ** 4:.1f}T"
+
+    def _get_type(self, vmlinux, block_device, gendisk):
```

Comment:
Again, type information and ideally context first and simpler parameters where possible afterwards.